api/src/routes/cart.ts

```diff
@@ -0,0 +1,321 @@
+ *         description: Cart or cart item not found
+ *   delete:
+ *     summary: Remove item from cart
+ *     tags: [Cart]
+ *     parameters:
+ *       - in: path
+ *         name: sessionId
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: Session identifier for the cart
+ *       - in: path
+ *         name: cartItemId
+ *         required: true
+ *         schema:
+ *           type: integer
+ *         description: Cart item ID
+ *     responses:
+ *       204:
+ *         description: Item removed from cart successfully
+ *       404:
+ *         description: Cart or cart item not found
+ */
+
+import express from 'express';
+import { getCartRepository } from '../repositories/cartRepo';
+import { NotFoundError, ValidationError } from '../utils/errors';
+
+const router = express.Router();
+
+// Get or create cart with items
+router.get('/:sessionId', async (req, res, next) => {
+  try {
+    const { sessionId } = req.params;
+    const repo = await getCartRepository();
+
+    // Get existing cart or create new one
+    let cart = await repo.getCartBySessionId(sessionId);
+    if (!cart) {
+      cart = await repo.createCart(sessionId);
+    }
+
+    // Get cart items with product details
+    const items = await repo.getCartItems(cart.cartId);
+
+    res.json({
+      cart,
+      items,
+    });
+  } catch (error) {
+    next(error);
+  }
+});
+
+// Add item to cart
+router.post('/:sessionId/items', async (req, res, next) => {
+  try {
+    const { sessionId } = req.params;
+    const { productId, quantity } = req.body;
+
+    // Validate input
+    if (!productId || !quantity) {
+      throw new ValidationError('productId and quantity are required');
+    }
+
+    if (typeof productId !== 'number' || typeof quantity !== 'number') {
+      throw new ValidationError('productId and quantity must be numbers');
+    }
+
+    if (quantity < 1) {
+      throw new ValidationError('quantity must be at least 1');
+    }
+
+    const repo = await getCartRepository();
+
+    // Get or create cart
+    let cart = await repo.getCartBySessionId(sessionId);
+    if (!cart) {
+      cart = await repo.createCart(sessionId);
+    }
+
+    // Add item to cart
+    const cartItem = await repo.addItem(cart.cartId, productId, quantity);
+
+    // Update cart timestamp
+    await repo.touchCart(cart.cartId);
+
+    res.status(201).json(cartItem);
+  } catch (error) {
+    next(error);
+  }
+});
+
+// Update cart item quantity
+router.put('/:sessionId/items/:cartItemId', async (req, res, next) => {
+  try {
+    const { sessionId, cartItemId } = req.params;
+    const { quantity } = req.body;
+
+    // Validate input
+    if (!quantity) {
+      throw new ValidationError('quantity is required');
+    }
+
+    if (typeof quantity !== 'number') {
+      throw new ValidationError('quantity must be a number');
+    }
+
+    if (quantity < 1) {
+      throw new ValidationError('quantity must be at least 1');
```

Comment:
Similar to the POST endpoint, the validation doesn't check if `quantity` is a positive integer. This could allow decimals, Infinity, or NaN to pass validation. Add integer validation:
```typescript
if (!Number.isInteger(quantity) || quantity < 1) {
  throw new ValidationError('quantity must be a positive integer');
}
```
```suggestion
    if (!Number.isInteger(quantity) || quantity < 1) {
      throw new ValidationError('quantity must be a positive integer');
```